pkg/render/intrusion_detection.go

```diff
@@ -1146,14 +1184,252 @@ func (c *intrusionDetectionComponent) intrusionDetectionAnnotations() map[string
+				APIGroups: []string{
+					"authentication.k8s.io",
+				},
+				Resources: []string{
+					"tokenreviews",
+					"subjectaccessreviews",
+				},
+				Verbs: []string{
+					"create",
+				},
+			},
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) getADAPIAccessRoleBinding() *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		TypeMeta: metav1.TypeMeta{Kind: "RoleBinding", APIVersion: "rbac.authorization.k8s.io/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     ADAPIObjectName,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      "ServiceAccount",
+				Name:      ADAPIObjectName,
+				Namespace: IntrusionDetectionNamespace,
+			},
+		},
+	}
+}
+
+// AD API Service and Deployment
+func (c *intrusionDetectionComponent) getADAPIService() *corev1.Service {
+	return &corev1.Service{
+		TypeMeta: metav1.TypeMeta{Kind: "Service", APIVersion: "v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{
+				"k8s-app": ADAPIObjectName,
+			},
+			Ports: []corev1.ServicePort{
+				{
+					Name: ADAPIObjectPortName,
+					Port: adAPIPort,
+					TargetPort: intstr.IntOrString{
+						Type:   intstr.Int,
+						IntVal: adAPIPort,
+					},
+					Protocol: corev1.ProtocolTCP,
+				},
+			},
+			Type: corev1.ServiceTypeClusterIP,
+		},
+	}
+}
+
+func (c *intrusionDetectionComponent) getADAPIDeployment() *appsv1.Deployment {
+	adAPIStorageVolumePath := "/storage"
+	adAPIStorageVolumeName := "volume-storage"
+	envVars := []corev1.EnvVar{
+		{Name: "LOG_LEVEL", Value: "info"},
+		{Name: "STORAGE_PATH", Value: adAPIStorageVolumePath},
+		{Name: "TLS_KEY", Value: c.cfg.ADAPIServerCertSecret.VolumeMountKeyFilePath()},
+		{Name: "TLS_CERT", Value: c.cfg.ADAPIServerCertSecret.VolumeMountCertificateFilePath()},
+	}
+
+	return &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "apps/v1"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ADAPIObjectName,
+			Namespace: IntrusionDetectionNamespace,
+			Labels: map[string]string{
+				"k8s-app": ADAPIObjectName,
+			},
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &adAPIReplicas,
+			Strategy: appsv1.DeploymentStrategy{
+				Type: appsv1.RecreateDeploymentStrategyType,
+			},
+			Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"k8s-app": ADAPIObjectName}},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      ADAPIObjectName,
+					Namespace: IntrusionDetectionNamespace,
+					Labels: map[string]string{
+						"k8s-app": ADAPIObjectName,
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: ADAPIObjectName,
+					Tolerations:        append(c.cfg.Installation.ControlPlaneTolerations, rmeta.TolerateMaster),
```

Comment:
Could you please add a small comment in the code as to why we want this toleration?

Reply:
thought this was a default config to add all our components in the control plane / master node, but now that I think about it would it be a good idea not to add this toleration since it needs disk space using `emptyDir` and might be better that the pod spawns on a worker node @doublek forgot if we chatted about this

Reply:
chatted with @doublek deployment should respect control plane node but not master 

https://github.com/tigera/operator/pull/1931/commits/1d1cd5c744eda16cb0ba70b7220849402aab16d0 updated with comments

Reply:
I wonder if I provided the wrong guidance by suggestion using ControlPlaneSelectors? I thought the intention of these were to make sure Tigera components run on the right nodes. If I've misunderstood the intent I am okay to revert this.